phpspy_trace.c

```diff
@@ -111,18 +111,20 @@ static int varpeek_find(trace_context_t *context, varpeek_entry_t *varpeek_map,
     memset(&zop, 0, sizeof(zop));
     try_copy_proc_mem("opline", (void*)local_execute_data->opline, &zop, sizeof(zop));
 
-    snprintf(varpeek_key, sizeof(varpeek_key), "%.*s:%d", file_len, file, zop.lineno); /* TODO support line ranges */
+    snprintf(varpeek_key, sizeof(varpeek_key), "%.*s", file_len, file);
     HASH_FIND_STR(varpeek_map, varpeek_key, varpeek);
-    if (!varpeek) return 0;
+    if (!varpeek || !((zop.lineno - varpeek->line_start) <= (varpeek->line_end - varpeek->line_start))) return 0;
 
     for (i = 0; i < op_array->last_var; i++) {
         try_copy_proc_mem("var", op_array->vars + i, &zstrp, sizeof(zstrp));
         try(rv, copy_zstring(context, "var", zstrp, tmp, sizeof(tmp), &tmp_len));
         if (strncmp(tmp, varpeek->varname, tmp_len) != 0) continue;
         try_copy_proc_mem("zval", ((zval*)(remote_execute_data)) + ((int)(5 + i)), &zv, sizeof(zv));
         try(rv, copy_zval(context, &zv, tmp, sizeof(tmp), &tmp_len));
+        if (strncmp(tmp, varpeek->varname, tmp_len) == 0) continue;
```

Comment:
i noticed that sometimes the trace prints the symbol instead of the value (fails to copy), this fixes that but there is likely a more efficient solution. we can just use two variables instead of reusing `tmp`